modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/db/IgnitePdsWithTtlExpirationOnDeactivateTest.java

```diff
@@ -0,0 +1,178 @@
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        final IgniteConfiguration cfg = super.getConfiguration(igniteInstanceName);
+
+        DataRegionConfiguration dfltRegion = new DataRegionConfiguration()
+                .setMaxSize(100 * 1024 * 1024)
+                .setPersistenceEnabled(true);
+
+        // Setting MaxWalArchiveSize to a relatively small value leads to frequent checkpoints (too many WAL segments).
+        cfg.setDataStorageConfiguration(
+            new DataStorageConfiguration()
+                    .setWalSegmentSize(8 * 1024 * 1024)
+                    .setMaxWalArchiveSize(16 * 1024 * 1024)
+                    .setCheckpointFrequency(10_000)
+                    .setDefaultDataRegionConfiguration(dfltRegion)
+                    .setWalMode(WALMode.LOG_ONLY));
+
+        cfg.setCacheConfiguration(
+            getCacheConfiguration(CACHE_NAME_ATOMIC).setAtomicityMode(ATOMIC)
+        );
+
+        return cfg;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected FailureHandler getFailureHandler(String igniteInstanceName) {
+        return new NoOpFailureHandler() {
+            @Override protected boolean handle(Ignite ignite, FailureContext failureCtx) {
+                failureHndTriggered = true;
+
+                return super.handle(ignite, failureCtx);
+            }
+        };
+    }
+
+    /**
+     * Returns a new cache configuration with the given name and {@code GROUP_NAME} group.
+     *
+     * @param name Cache name.
+     * @return Cache configuration.
+     */
+    private CacheConfiguration<?, ?> getCacheConfiguration(String name) {
+        CacheConfiguration<?, ?> ccfg = new CacheConfiguration<>();
+
+        ccfg.setName(name);
+        ccfg.setExpiryPolicyFactory(AccessedExpiryPolicy.factoryOf(new Duration(TimeUnit.MILLISECONDS, EXPIRATION_TIMEOUT)));
+        ccfg.setEagerTtl(true);
+
+        return ccfg;
+    }
+
+    /** */
+    @Test
+    public void testStartAfterDeactivateWithTtlExpiring() throws Exception {
+        IgniteEx srv = startGrid(0);
+
+        srv.cluster().state(ClusterState.ACTIVE);
+
+        IgniteCache<Integer, String> cache = srv.cache(CACHE_NAME_ATOMIC);
+
+        AtomicBoolean timeoutReached = new AtomicBoolean(false);
+
+        AtomicInteger threadId = new AtomicInteger(0);
+
+        IgniteInternalFuture<?> ldrFut = runMultiThreadedAsync(() -> {
+            int id = threadId.getAndIncrement();
+
+            int i = 0;
+            while (!timeoutReached.get()) {
+                cache.put(id * 1_000_000 + i, RandomStringUtils.randomAlphanumeric(1000));
+                i++;
+            }
+        }, WORKLOAD_THREADS_CNT, "loader");
+
+        doSleep(EXPIRATION_TIMEOUT * 2);
+        timeoutReached.set(true);
+        ldrFut.get();
+
+        srv.cluster().state(INACTIVE);
+
+        stopGrid(0);
+
+        startGrid(0);
+
+        awaitPartitionMapExchange();
```

Comment:
It is not obvious why we need this call for a cluster of 1 node.

Reply:
Great, thank you, forgot to remove this (remains of experiments, you know :))